usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -621,12 +622,19 @@ def calculate_complex_fields(self, row, hit):
                 "county_name": hit.get("sub_recipient_location_county_name"),
                 "address_line1": hit.get("sub_recipient_location_address_line1"),
                 "congressional_code": hit.get("sub_recipient_location_congressional_code"),
-                "zip4": hit.get("sub_recipient_location_zip")[5:],
+                "zip4": zip4,
                 "zip5": hit.get("sub_recipient_location_zip5"),
                 "foreign_postal_code": hit.get("sub_recipient_location_foreign_posta"),
             }
 
         if "Sub-Award Primary Place of Performance" in self.fields:
+            if hit.get("sub_pop_zip"):
+                zip4 = hit.get("sub_pop_zip")[5:]
```

Comment:
Are all of these going to be 9 digits?

Reply:
I looked into this. In production we have 221,520 Subawards in Elasticsearch where `sub_pop_zip` is present, but less than 9 digits. I've updated the logic to account for these zips.